lgc/patch/NggPrimShader.h

```diff
@@ -355,9 +353,35 @@ class NggPrimShader {
     llvm::Value *vertexIndex0; // Relative index of vertex0 in NGG subgroup
     llvm::Value *vertexIndex1; // Relative index of vertex1 in NGG subgroup
     llvm::Value *vertexIndex2; // Relative index of vertex2 in NGG subgroup
-
   } m_nggInputs = {};
 
+  // ES handlers
+  struct {
+    llvm::Function *esMain; // ES main function
+    // A pair of part ES after ES splitting: the first one is to fetch cull data (position and cull distance) and the
+    // second one is to do deferred vertex exporting
+    std::pair<llvm::Function *, llvm::Function *> partEs;
+  } m_esHandlers = {};
+
+  // GS handlers
+  struct {
+    llvm::Function *gsMain;     // GS main function
+    llvm::Function *copyShader; // Copy shader
+    llvm::Function *gsEmit;     // GS emit handler
+    llvm::Function *gsCut;      // GS cut handler
+  } m_gsHandlers = {};
```

Comment:
Could remove the "gs" prefixes since it's already part of the containing struct name. Same for ES, actually.

Reply:
No problem.